routing/pathfind.go

```diff
@@ -365,6 +357,24 @@ func findPath(g *graphParams, r *RestrictParams, cfg *PathFindingConfig,
 		}
 	}
 
+	numNodes, err := g.graph.NumNodes(tx)
+	if err != nil {
+		return nil, err
+	}
+
+	// First we'll initialize an empty heap which'll help us to quickly
+	// locate the next edge we should visit next during our graph
+	// traversal.
+	nodeHeap := newDistanceHeap(numNodes)
```

Comment:
instead of fetching the exact number of nodes in the db, could we make an educated constant guess on this number? Avoids the extra db code, and I don't think it matters since we don't know if it will be filled anyway.

Reply:
what way of making that guess do you have in mind?

Reply:
For what it's worth, counting from the DB caused little to no impact in the benchmarks. Since it's only using the index it's most likely already cached in memory. I tried to use some boltd guesstimates but they were really far from the real number (like number of leafs).

Reply:
The previous commit removed the graph traversal, and this reeintroduces it. This is probably less expensive, since we only count, but I wonder if we would get the same effect from just setting `numNodes = 1000` for instance.

Reply:
Discussed with @halseth offline. Doubts remain whether the perf. improvement is worth the extra db scan through all nodes. Maybe a good middle ground is to add a reasonable constant now. For example 10000 nodes. This is little extra memory, because it is only a list of pointers, and it should do the job too. Then, when the cache is being used, we can possibly take a count from the cache. That way, we'll have a better estimate after a few runs.

Reply:
Sure, I'll add a constant for this. The only real gain with having this call is a bit of "future proofing".